test/daemonv2/daemon.go

```diff
@@ -0,0 +1,156 @@
+package daemonv2
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"syscall"
+	"time"
+
+	daemon "github.com/alibaba/pouch/daemon/config"
+	"github.com/alibaba/pouch/test/command"
+	"github.com/alibaba/pouch/test/util"
+
+	"github.com/gotestyourself/gotestyourself/icmd"
+)
+
+// New returns the daemon struct.
+func New() *Daemon {
+	return &Daemon{
+		Config: daemon.Config{
+			Listen:         []string{Listen},
+			HomeDir:        HomeDir,
+			ContainerdAddr: ContainerdAdd,
+			Pidfile:        Pidfile,
+			Debug:          true,
+		},
+		Bin:        PouchdBin,
+		LogPath:    DaemonLog,
+		ConfigJSON: ConfigJSON,
+
+		timeout: 15,
+	}
+}
+
+// IsUp checks the up or not
+func (d *Daemon) IsUp() bool {
+	return command.PouchRun("--host", d.Config.Listen[0], "version").ExitCode == 0
+}
+
+// Clean kills pouch daemon and clean config file
+func (d *Daemon) Clean() {
+	d.Stop()
+
+	if d.ConfigJSON != "" {
+		os.Remove(d.ConfigJSON)
+	}
+}
+
+// DumpLog prints the daemon log
+func (d *Daemon) DumpLog() {
+	d.LogFile.Sync()
+
+	content, err := ioutil.ReadFile(d.LogPath)
+	if err != nil {
+		fmt.Printf("failed to read log, err: %s\n", err)
+	}
+	fmt.Printf("pouch daemon log contents:\n %s\n", content)
+}
+
+// Restart restarts pouch daemon
+func (d *Daemon) Restart() error {
+	if err := d.Stop(); err != nil {
+		return err
+	}
+
+	return d.Start()
+}
+
+// Stop stops pouch daemon
+func (d *Daemon) Stop() error {
+	if !d.IsUp() {
+		return nil
+	}
+
+	if d.Pid != 0 {
+		// kill pouchd and all other process in its group
+		err := syscall.Kill(-d.Pid, syscall.SIGKILL)
+		if err != nil {
+			fmt.Printf("kill pouchd failed, err:%s", err)
```

Comment:
I think we can use `pkgerrors.Wrap` and return error. Then caller will check or log the error. What do you think?